src/components/PlantsPrimaryPage/index.tsx

```diff
@@ -0,0 +1,184 @@
+import { Select } from '@terraware/web-components';
+import { PlantingSite } from 'src/types/Tracking';
+import { useDeviceInfo } from '@terraware/web-components/utils';
+import { useHistory, useParams } from 'react-router-dom';
+import useSnackbar from 'src/utils/useSnackbar';
+import { PreferencesService, TrackingService } from 'src/services';
+import PageHeaderWrapper from 'src/components/common/PageHeaderWrapper';
+import PageSnackbar from 'src/components/PageSnackbar';
+import { useOrganization } from 'src/providers/hooks';
+
+export type PlantsPrimaryPageProps = {
+  title: string;
+  children: React.ReactNode; // primary content for this page
+  onSelect: (plantingSite: PlantingSite) => void; // planting site selected, id of -1 refers to All
+  pagePath: string;
+  lastVisitedPreferenceName: string;
+  plantsSitePreferences?: Record<string, unknown>;
+  setPlantsSitePreferences: (preferences: Record<string, unknown>) => void;
+  allowAllAsSiteSelection?: boolean; // whether to support 'All' as a planting site selection
+};
+
+const allSitesOption = () => ({
+  name: strings.ALL,
+  id: -1,
+});
+
+export default function PlantsPrimaryPage({
+  title,
+  children,
+  onSelect,
+  pagePath,
+  lastVisitedPreferenceName,
+  plantsSitePreferences,
+  setPlantsSitePreferences,
+  allowAllAsSiteSelection,
+}: PlantsPrimaryPageProps): JSX.Element {
+  const { selectedOrganization } = useOrganization();
+  const [selectedPlantingSite, setSelectedPlantingSite] = useState<PlantingSite>();
+  const [plantingSites, setPlantingSites] = useState<PlantingSite[]>();
+  const theme = useTheme();
+  const { isMobile } = useDeviceInfo();
+  const { plantingSiteId } = useParams<{ plantingSiteId: string }>();
+  const history = useHistory();
+  const snackbar = useSnackbar();
+  const contentRef = useRef(null);
+
+  useEffect(() => {
+    if (plantsSitePreferences) {
+      PreferencesService.updateUserOrgPreferences(selectedOrganization.id, {
+        [lastVisitedPreferenceName]: plantsSitePreferences,
+      });
+    }
+  }, [plantsSitePreferences, lastVisitedPreferenceName, selectedOrganization.id]);
+
+  useEffect(() => {
+    const populatePlantingSites = async () => {
+      const serverResponse = await TrackingService.listPlantingSites(selectedOrganization.id);
+      if (serverResponse.requestSucceeded) {
+        if (allowAllAsSiteSelection && serverResponse.sites?.length) {
+          setPlantingSites([allSitesOption(), ...serverResponse.sites]);
+        } else {
+          setPlantingSites(serverResponse.sites ?? []);
+        }
+      } else {
+        snackbar.toastError();
+      }
+    };
+    populatePlantingSites();
+  }, [selectedOrganization.id, snackbar, allowAllAsSiteSelection]);
+
+  const setActivePlantingSite = useCallback(
+    (site: PlantingSite | undefined) => {
+      if (site) {
+        history.push(pagePath.replace(':plantingSiteId', site.id.toString()));
```

Comment:
Should this be a `replace` type of history update? Do we want the user to go to the sequence of last selected planting sites when they hit the back button?

Reply:
Yes we do want them to go back. The selections are url driven today.